detectable/src/main/java/com/synopsys/integration/detectable/detectables/maven/cli/MavenCodeLocationPackager.java

```diff
@@ -187,21 +202,38 @@ private void populateGraphDependencies(ExcludedIncludedWildcardFilter scopeFilte
                 dependencyParentStack.push(dependency);
             }
         }
+        if(!shadedDependenciesConverted.isEmpty() && shadedDependenciesConverted.containsKey(dependency.getExternalId())) {
+            for(Dependency childDependency: shadedDependenciesConverted.get(dependency.getExternalId())) {
+                currentGraph.addParentWithChild(dependency, childDependency);
+            }
+            shadedDependenciesConverted.remove(dependency.getExternalId());
+        }
     }
 
     private void addOrphansToGraph(DependencyGraph graph, List<Dependency> orphans) {
         logger.trace(String.format("# orphans: %d", orphans.size()));
-        if (orphans.size() > 0) {
+        if (!orphans.isEmpty() || !shadedDependenciesConverted.isEmpty()) {
             Dependency orphanListParent = createOrphanListParentDependency();
             logger.trace(String.format("adding orphan list parent dependency: %s", orphanListParent.getExternalId().toString()));
             graph.addChildToRoot(orphanListParent);
             for (Dependency dependency : orphans) {
                 logger.trace(String.format("adding orphan: %s", dependency.getExternalId().toString()));
                 graph.addParentWithChild(orphanListParent, dependency);
             }
+            addShadedDependenciesToGraph(graph, orphanListParent);
```

Comment:
PI might send us some shaded dependencies which are part of transitives of a transitive dependency that might not be present in the tree ( dependency could come from jar files as PI scans mirrors, proxies and repositories for finding dependencies). But to be consistent with PI we just add those dependencies as additional components in BOM.